CMakeLists.txt

```diff
@@ -1,72 +1,27 @@
-#
-# TBD:
-#  * jemalloc support
-#  * count
-#
-# Notes:
-#  * Use the -G"Visual Studio 14 2015 Win64" generator on Windows. Without this
-#    a 32-bit project will be created. There is no way to set the generator or
-#    force a 64-bit build in CMakeLists.txt (setting CMAKE_GENERATOR_PLATFORM won't work).
-#    The best solution may be to wrap cmake with a script.
-#
-#  * It is not possible to generate a visual studio project on linux or
-#    mac. The visual studio generator is only available on windows.
-#
-#  * Language is _much_ worse than python, poor documentation and "quirky"
-#    language support (for example, generator expressions can only be used
-#    in limited contexts and seem to work differently based on
-#    context (set_property can set multiple values, add_compile_options
-#    can not/or is buggy)
-#
-#  * Could not call out to `sed` because cmake messed with the regular
-#    expression before calling the external command. I did not see a way
-#    around this.
-#
-#  * Makefile generators want to be single target. It wants a separate
-#    directory for each target type. I saw some mentions on the web for
-#    ways around this bug haven't look into it. The visual studio project
-#    does support debug/release configurations in the same project (but
-#    not unity/non-unity).
 
-############################################################
+cmake_minimum_required(VERSION 3.9)
+project(ripple-libppdemo)
 
-#########################################################
-# CMAKE_C_COMPILER and CMAKE_CXX_COMPILER must be defined
-# before the project statement; However, the project
-# statement will clear CMAKE_BUILD_TYPE. CACHE variables,
-# along with the order of this code, are used to work
-# around these constraints.
-#
-# Don't put any code above or in this block, unless it
-# has similar constraints.
-cmake_minimum_required(VERSION 3.1.0)
-list(APPEND CMAKE_MODULE_PATH
-  "${CMAKE_CURRENT_SOURCE_DIR}/extras/rippled/Builds/CMake")
-include(CMakeFuncs)
-set(openssl_min 1.0.2)
-parse_target()
-project(ripple-libpp)
-#########################################################
+#[===========================================[
+  this pulls-in the rippled source code via
+  git submodule. This demo exe depends on the
+  xrpl_core library which is defined by the
+  rippled project.
+#]===========================================]
+if (MSVC)
+  #[=======================================[
+    HACK to patch a one-line windows-only
+    bug in rippled CMakeLists.txt. This can
+    be removed once rippled 1.2 is available
+    and we update the submodule
+  #]=======================================]
+  execute_process( COMMAND powershell -Command
+     "(Get-Content extras/rippled/CMakeLists.txt).replace('exclude_if_included (rocksdb)', '##') \
+     | Set-Content extras/rippled/CMakeLists.txt"
+     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  )
+endif ()
```

Comment:
I'd expect a new beta soon. What do you think about holding up this PR so we can remove this hack?

Reply:
I think the submodule points to `master` (as it should), which is more like a 3 month wait...which I think is still fine if we want to hold it.

Reply:
I vote to merge with the hack:
1. The patch only applies to one environment (Windows)
2. Once the `rippled` fix is merged to master, the affected line will (presumably) be fixed to be in a block that is only relevant in non-Windows. (And if it doesn't get merged, then the hack stays relevant.)
3. Because the patch modifies a git tracked file, I think anybody who does a Windows built is likely to notice as soon as they run `git status` or even sooner if they use one of the various git aware shell prompts.

Reply:
I'm fine with that.